src/h5web/vis-packs/core/utils.ts

```diff
@@ -262,3 +271,65 @@ export function isScaleType(val: unknown): val is ScaleType {
     typeof val === 'string' && Object.values<string>(ScaleType).includes(val)
   );
 }
+
+export function getCombinedDomain(
+  domain: Domain | undefined,
+  domainsToCombine: (Domain | undefined)[]
+): Domain | undefined {
+  if (domainsToCombine.length === 0) {
+    return domain;
+  }
+
+  const [domainToCombine, ...otherDomains] = domainsToCombine;
+
+  if (domain === undefined || domainToCombine === undefined) {
+    return domain || domainToCombine;
+  }
+
+  return getCombinedDomain(
+    [
+      Math.min(domain[0], domainToCombine[0]),
+      Math.max(domain[1], domainToCombine[1]),
+    ],
+    otherDomains
+  );
+}
+
+export function getBaseArray<T extends unknown[] | undefined>(
+  value: T,
+  rawDims: number[]
+): T extends (infer U)[] ? ndarray<U> : undefined;
```

Comment:
I wonder whether we still need those override declarations, since `useMappedArray` has one already. Yeah... probably for `useMappedArrays`... Might be worth trying to remove the override declaration on `useMappedArray, actually. Or maybe you've already tried and didn't get anywhere. Not a merge blocker either way.

Reply:
Removing the override of `useMappedArray` works :fishing_pole_and_fish:

Reply:
Hmm still seems to be there, maybe you forgot to push?

Reply:
Hm I don't know what I imagined but removing it doesn't work in fact.

It might be because we return either `[ndarray, ndarray]` or `[undefined, undefined]` which is interpreted as `(ndarray | undefined)[]` without the override.